votor/src/certificate_pool_service.rs

```diff
@@ -157,19 +158,38 @@ impl CertificatePoolService {
                     &mut events,
                     &ctx.commitment_sender,
                 ) {
-                    Ok(Some(finalized_slot)) => {
-                        // Reset standstill timer
-                        debug_assert!(finalized_slot > highest_finalized_slot);
-                        highest_finalized_slot = finalized_slot;
-                        standstill_timer = Instant::now();
-                        // Set root
-                        let root_bank = ctx.root_bank_cache.root_bank();
-                        if root_bank.slot() > current_root {
-                            current_root = root_bank.slot();
-                            cert_pool.handle_new_root(root_bank);
+                    Ok((new_finalized_slot, new_certificates_to_send)) => {
```

Comment:
could we move this entire match statement to a helper function that returns a result?
Then if the helper function returns an error corresponding to the bls_sender being disconnected, or AddVoteError::CertificateSendError we can shutdown, if it's a "unable to push vote into pool" type of error we can `continue`

Then in a future PR we can also bubble this error all the way up to the `new` fn for shutting down #284